models/sessions/scratch/databricks/snowplow_web_sessions_this_run.sql

```diff
@@ -0,0 +1,363 @@
+{{
+    config(
+        tags=["this_run"]
+    )
+}}
+
+with my_firsts as (
+    select
+        -- app id
+        app_id as app_id,
+
+        -- session fields
+        domain_sessionid,
+        domain_sessionidx,
+
+        {{ snowplow_utils.current_timestamp_in_utc() }} as model_tstamp,
+
+        -- user fields
+        user_id,
+        domain_userid,
+        {% if var('snowplow__session_stitching') %}
+            -- updated with mapping as part of post hook on derived sessions table
+            cast(domain_userid as {{ snowplow_utils.type_string(255) }}) as stitched_user_id,
+        {% else %}
+            cast(null as {{ snowplow_utils.type_string(255) }}) as stitched_user_id,
+        {% endif %}
+        network_userid as network_userid,
+
+        -- first page fields
+        page_title as first_page_title,
+        page_url as first_page_url,
+        page_urlscheme as first_page_urlscheme,
+        page_urlhost as first_page_urlhost,
+        page_urlpath as first_page_urlpath,
+        page_urlquery as first_page_urlquery,
+        page_urlfragment as first_page_urlfragment,
+
+        -- referrer fields
+        page_referrer as referrer,
+        refr_urlscheme as refr_urlscheme,
+        refr_urlhost as refr_urlhost,
+        refr_urlpath as refr_urlpath,
+        refr_urlquery as refr_urlquery,
+        refr_urlfragment as refr_urlfragment,
+        refr_medium as refr_medium,
+        refr_source as refr_source,
+        refr_term as refr_term,
+
+        -- marketing fields
+        mkt_medium as mkt_medium,
+        mkt_source as mkt_source,
+        mkt_term as mkt_term,
+        mkt_content as mkt_content,
+        mkt_campaign as mkt_campaign,
+        mkt_clickid as mkt_clickid,
+        mkt_network as mkt_network,
+
+        -- Most if not all the following fields should be the same across all events in a session, but this ensures they are
+        -- geo fields
+        geo_country as geo_country,
+        geo_region as geo_region,
+        geo_region_name as geo_region_name,
+        geo_city as geo_city,
+        geo_zipcode as geo_zipcode,
+        geo_latitude as geo_latitude,
+        geo_longitude as geo_longitude,
+        geo_timezone as geo_timezone,
+
+        -- ip address
+        user_ipaddress as user_ipaddress,
+
+        -- user agent
+        useragent as useragent,
+
+        br_renderengine as br_renderengine,
+        br_lang as br_lang,
+        os_timezone as os_timezone,
+
+        -- optional fields, only populated if enabled.
+        -- iab enrichment fields
+        {% if var('snowplow__enable_iab', false) %}
```

Comment:
Should we consider making this a macro? Same for the other (non-BQ) warehouses

Reply:
I'm of two minds, making it a macro means it can be easily re-used across the different warehouses (although merging the snowflake and databricks sql would be messy and break the clean folder structure) so more similar and shorter sql, and could also be used in page views. On the other hand it obfuscates the code and makes it harder to follow the column lineage like with BigQuery. 

Up to you, happy either way

Reply:
Made up my mind, did it